async-opcua-client/src/session/connection.rs

```diff
@@ -116,59 +147,84 @@ impl<T, R> SessionBuilder<'_, T, R> {
                 .is_some_and(|e| e.iter().any(|p| p.token_type == UserTokenType::IssuedToken)),
         }
     }
+
+    /// Set the connection source to use. This is used to create the transport
+    /// connector. Defaults to a direct TCP connection, implemented by `()`.
+    pub fn with_connector<C2>(self, connection_source: C2) -> SessionBuilder<'a, T, R, C2>
```

Comment:
Just a tiny little suggestion:

```suggestion
    pub fn with_connector<CS>(self, connection_source: CS) -> SessionBuilder<'a, T, R, CS>
```